testsuite/single/tests/efuns/read_file.c

```diff
@@ -20,11 +20,9 @@ void do_tests() {
     ASSERT(!read_file("/testfile", 10000, 1));
     
     // Start tests for end of file -lines
-    mid = implode(explode(foo, "\n")[<14..<5], "\n") + "\n";
-    ASSERT_EQ(mid, read_file("/testfile", -15, 10));
+    ASSERT_EQ(implode(explode(foo, "\n")[<15..<6], "\n") + "\n", read_file("/testfile", -15, 10));
     // if we start too many lines before the end of the file, start at the beginning
     ASSERT(read_file("/testfile", -10000, 1));
-
     // CRLF tests
     ASSERT(read_file("/crlf_test_file.lf") == read_file("/crlf_test_file.crlf"));
```

Comment:
Can you add a test case for negative reading using test file.crlf ?

Reply:
Well this is only testing out-of-range cases, I think we need to cover normal cases as well, the same case as above.

It should work, but i think we might need a test.